lib/connection/connection.js

```diff
@@ -205,6 +209,47 @@ var closeHandler = function(self) {
   }
 }
 
+var decompress = function(compressorId, compressedData, callback) {
+  var decompressors = [
+    function(compressedData, callback) {
+      callback(null, compressedData)
+    },
+    Snappy.uncompress,
+    zlib.unzip
+  ];
+
+  if (compressorId < 0 || compressorId > decompressors.length - 1) {
+    throw new Error('Server sent message compressed using an unsupported compressor');
+  }
+
+  // Pass the compressed date to the appropriate decompressor
+  decompressors[compressorId](compressedData, function(err, decompressedData) {
+    if (err) {
+      console.log(err)
```

Comment:
two issues:
  - you're using `console.log` here, instead of the logger
  - when the error occurs you should be calling back with `callback(err, null)` and returning. Presently you just print out the error, and then return null/undefined value for a success path.